### DISCUSSION TITLE
Can Saleor handle 1.5 million products and 30,000 categories with a category depth of 6?

### DISCUSSION THREAD

**Discussion by PhilippeKev** [saleor/saleor#12346] [General]
Hello Saleor team,

I am considering using Saleor for my e-commerce project, and I would like to know if it can handle my specific use case. I have around 1.5 million products distributed across nearly 30,000 different categories. The depth of the category structure is 5 levels deep.

My questions are:

    Can Saleor efficiently handle this large number of products and categories without any significant performance issues?
    Are there any known limitations or issues when dealing with such a large and complex category structure in Saleor?
    Are there any recommendations or best practices for managing this scale of products and categories with Saleor?

I would greatly appreciate your insights and advice on whether Saleor is suitable for this use case. Thank you for your time and assistance.

Best regards,

Philippe

**Comment by karolkielecki** [comment]
Hi @PhilippeKev, yes - Saleor is battle-tested with a large number of products and categories. 

Some of our Cloud users built integrations with PIM tools to manage many products. You can find more info on how to build an app in our docs: https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts

Performance depends on many factors. Have you considered using [Saleor Cloud](https://cloud.saleor.io/signup)? You can create a free developer account to test it.

**Reply by PhilippeKev** [reply]
Thank you for your responses, @karolkielecki. I appreciate the information about Saleor being tested with a large number of products and categories. However, I still have some concerns regarding the depth of the category structure.

In my use case, the category structure is 5 levels deep, while I believe Saleor might be limited to a depth of 3. Can you please confirm if Saleor can support a category depth of 5 levels without any issues? If it's not natively supported, are there any workarounds or customizations you could recommend to handle a deeper category structure in Saleor?

I appreciate your help in clarifying this concern.

Best regards,

**Reply by karolkielecki** [reply]
Hi @PhilippeKev, could you please write more details about your use case - how exactly categories should work, etc.? An example of a store would help us understand your requirements.

**Reply by PhilippeKev** [reply]
Thank you for your response, @karolkielecki. I'll provide more details about my use case and the category structure to help you better understand my requirements.

In my e-commerce store, I have a wide range of products related to collectibles and memorabilia, specifically postcards. The products are organized into categories with a hierarchy depth of 6 levels. Here's an example of the category structure:

    Postcards
        1.1. Africa
        1.2. America
        1.3. Asia
        1.4. Europe
            1.4.1. Belgium
            1.4.2. France
                1.4.2.1. Regions
                1.4.2.2. [01] Ain
                ...
                1.4.2.3. [75] Paris
                    1.4.2.3.1. Unclassified
                    1.4.2.3.2. Panoramic views
                    1.4.2.3.3. Arc de Triomphe
                    1.4.2.3.4. Champs-Elysées
                    1.4.2.3.5. Churches
                    1.4.2.3.6. Districts
                    - 1.4.2.3.6.1. District 01
                    - ...
                    - 1.4.2.3.6.2. District 20
                    1.4.2.3.7. Museums
                    1.4.2.3.8. Tour Eiffel
            1.4.3. Germany
            1.4.4. UK
        1.5. Oceania
        1.6. World
        1.7. Topics
    Papers
    Stamps
    Others

As you can see, the category structure goes up to 6 levels deep in some cases. The purpose of this structure is to provide a more granular and precise organization of the products, making it easier for customers to find what they are looking for.

My main concern is whether Saleor can support this level of category depth without any performance or usability issues. If Saleor's default functionality does not support a depth of 6, I would appreciate any suggestions or workarounds to customize Saleor to accommodate my use case.

You can find an example on this website : https://www.delcampe.net/en_GB/collectables/postcards/

Thank you for your help and guidance.

Best regards,

**Reply by timuric** [reply]
Hi @PhilippeKev Saleor supports much deeper level of category structure than 6

**Comment by timuric** [comment]
Hey @PhilippeKev Saleor is tested on millions of products and should handle your load gracefully

**Reply by PhilippeKev** [reply]
Thank you for your responses, @timuric. I appreciate the information about Saleor being tested with a large number of products and categories. However, I still have some concerns regarding the depth of the category structure.

In my use case, the category structure is 5 levels deep, while I believe Saleor might be limited to a depth of 3. Can you please confirm if Saleor can support a category depth of 5 levels without any issues? If it's not natively supported, are there any workarounds or customizations you could recommend to handle a deeper category structure in Saleor?

I appreciate your help in clarifying this concern.

Best regards,

**Reply by timuric** [reply]
There are no limits on category structure in Saleor

**Comment by michalmirumee** [comment]
Hi @PhilippeKev! This project scale looks impressive! Would you like to have a quick chat about it? We know Saleor quite well, would love to give a few hints. Maybe a call?

**Reply by PhilippeKev** [reply]
Hi @michalmirumee ,

Thank you for your offer to discuss my project further. I appreciate your willingness to help and provide guidance on using Saleor for my use case. At the moment, I would prefer to continue the conversation in writing, as it is easier for me to refer back to the information provided.

My primary concern is still the depth of the category structure, as it goes up to 6 levels deep. I would greatly appreciate your insights on whether Saleor can handle this category depth and if there are any specific configurations or customizations that may be required.

Additionally, if you have any other concerns or potential issues that I might not have identified related to using Saleor for a project of this scale (1.5 million products and 30,000 categories), please share your thoughts. Your expertise and experience with Saleor could provide valuable information for me to consider while planning my e-commerce platform.

Thank you again for your time and assistance.

Best regards,
